From 444c524670f9d160403de35856583459a923f661 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Tue, 10 Dec 2019 11:17:28 +0200
Subject: [PATCH 20/37] MLK-22378: ASoC: imx-wm8524: remove missing capture
 DAPM routes

commit 77597bde1626f6438dd63652ae3610e605762551 from
https://source.codeaurora.org/external/imx/linux-imx.git

wm8524 has no capture support, thus remove missing DAPM routes.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8524.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8524.c b/sound/soc/fsl/imx-wm8524.c
index a35b90e94f17..b94053d87f69 100644
--- a/sound/soc/fsl/imx-wm8524.c
+++ b/sound/soc/fsl/imx-wm8524.c
@@ -75,9 +75,7 @@ static struct snd_soc_ops imx_hifi_ops = {
 
 static const struct snd_soc_dapm_route audio_map[] = {
 	{"Playback",  NULL, "CPU-Playback"},
-	{"CPU-Capture",  NULL, "Capture"},
 	{"CPU-Playback",  NULL, "ASRC-Playback"},
-	{"ASRC-Capture",  NULL, "CPU-Capture"},
 };
 
 static int be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
@@ -193,7 +191,7 @@ static int imx_wm8524_probe(struct platform_device *pdev)
 	priv->card.num_dapm_widgets = ARRAY_SIZE(imx_wm8524_dapm_widgets);
 	priv->card.dai_link = priv->dai;
 	priv->card.dapm_routes = audio_map;
-	priv->card.num_dapm_routes = 2;
+	priv->card.num_dapm_routes = 1;
 
 	/*if there is no asrc controller, we only enable one device*/
 	if (asrc_pdev) {
@@ -220,7 +218,7 @@ static int imx_wm8524_probe(struct platform_device *pdev)
 		priv->dai[2].be_hw_params_fixup = be_hw_params_fixup,
 		priv->card.num_links = 3;
 		priv->card.dai_link = &priv->dai[0];
-		priv->card.num_dapm_routes += 2;
+		priv->card.num_dapm_routes += 1;
 
 		ret = of_property_read_u32(asrc_np, "fsl,asrc-rate",
 					   &priv->asrc_rate);
-- 
2.17.1

